node/consensus/src/lib.rs

```diff
@@ -234,10 +240,6 @@ impl<N: Network> Consensus<N> {
         {
             let transaction_id = transaction.id();
 
-            // Check that the transaction is not a fee transaction.
-            if transaction.is_fee() {
-                bail!("Transaction '{}' is a fee transaction {}", fmt_id(transaction_id), "(skipping)".dimmed());
-            }
```

Comment:
we may want to keep this for performance (this avoids one lock access and 2 lookups); also, we currently don't touch `seen_transactions` in case of this early return

Reply:
4a92b3a22